test/TritonGPU/combine.mlir

```diff
@@ -1056,8 +1056,9 @@ module attributes {"triton_gpu.num-warps" = 4 : i32} {
     %12 = tt.broadcast %11 : (tensor<1x1xi32, #blocked>) -> tensor<1x256xi32, #blocked>
     %13 = tt.splat %arg1 : (!tt.ptr<f32>) -> tensor<1x256x!tt.ptr<f32>, #blocked>
     %14 = tt.broadcast %7 : (tensor<1x1xi1, #blocked>) -> tensor<1x256xi1, #blocked>
-    %15 = scf.for %arg5 = %c0_i32 to %c3136_i32 step %c256_i32 iter_args(%arg6 = %cst) -> (tensor<1x256xf32, #blocked>)  : i32 {
-      %43 = tt.splat %arg5 : (i32) -> tensor<1x256xi32, #blocked>
+    %15 = scf.for %arg5 = %c0_i32 to %c3136_i32 step %c256_i32 iter_args(%arg6 = %cst) -> (tensor<1x256xf32, #blocked>) {
+      %42 = arith.index_cast %arg5 : index to i32
+      %43 = tt.splat %42 : (i32) -> tensor<1x256xi32, #blocked>
```

Comment:
This part of changes is not the goal of this PR. I just found it's not working and fixed it.